src/transforms/replace_enums_by_nat.coffee

```diff
@@ -10,24 +10,26 @@ do () =>
       switch root.constructor.name
         when "Scope"
           if root.original_node_type == "SourceUnit"
-            ctx.enums_map = {}
+            ctx.enums_map = new Map
```

Comment:
Why not switch to `new Set` in this case?
There would be no need to store `true` for every element

Reply:
I never use Set because you never know when you need to store some extra data with key. So I use Map always (even when it's not needed right now)

Reply:
![image](https://user-images.githubusercontent.com/4216701/87028945-0ba29800-c1e8-11ea-91e3-09c49c4a89de.png)
![image](https://user-images.githubusercontent.com/4216701/87028974-165d2d00-c1e8-11ea-8707-c6d46dfe6004.png)
But they are made to be almost fully swappable. You almost always can just switch And despite performance improvements `Set` better shows programmer's intention.